src/hidimstat/base_perturbation.py

```diff
@@ -92,7 +94,7 @@ def fit(self, X, y=None, groups=None):
         else:
             raise ValueError("groups needs to be a dictionnary")
 
-    def predict(self, X):
+    def predict(self, X, random_generator):
```

Comment:
```suggestion
    def predict(self, X):
```
Similar to #377, I think we should not have additional arguments in the `predict` method. why not using a `rng = check_random_state(self.random_state)` at the begening oft he method?